zulipterminal/cli/run.py

```diff
@@ -281,15 +281,23 @@ def main(options: Optional[List[str]]=None) -> None:
             theme_to_use[0] in available_themes
             or theme_to_use[0] in theme_aliases
         )
-        if not is_valid_theme:
-            print("Invalid theme '{}' was specified {}."
-                  .format(*theme_to_use))
+        if not is_valid_theme or args.list_themes:
+            exit_code = 0
+            if not is_valid_theme:
+                print("Invalid theme '{}' was specified {}."
+                      .format(*theme_to_use))
+                exit_code = 1
             print("The following themes are available:")
             for theme in available_themes:
-                print("  ", theme)
+                suffix = ""
+                if theme == "zt_dark":
+                    suffix += "[default theme]"
+                if theme_to_use[0] == theme:
+                    suffix += "[user preference]"
+                print("  ", theme, suffix)
```

Comment:
This doesn't get run for just `--list-themes` - there is still a block of code earlier in the file that is triggered earlier, which I see you've fixed in the last commit. I'd expected it in the same commit.

I think users may want to just run `zulip-term --list-themes` without necessarily having/specifying a config file, so I'm not sure we can actually move the code down here, though it would be good to have the same code called - so an ideal place to use a function. That would allow the user-preference to be optional, if we don't know it - if we don't have access to the file.

Reply:
I have removed this part and will commit it later.

Reply:
What code do you want to put in the new function?

Reply:
I would like `zulip-term --list-themes -c invalid` to still work, which requires some code to remain at the top, as it was before.

However, we also want to run the same type of code in the lower section, so to avoid duplication we can use a function to avoid duplication.

if we have extra information then the lower version can do a slightly different output. This could depend on a parameter to the new function - or depending if the function outputs the information or returns a data-structure, the data could be modified before output in the lower section instead.